src/trail.js

```diff
@@ -0,0 +1,260 @@
+import {path} from "d3-path";
+import constant from "./constant.js";
+import {x as pointX, y as pointY, z as pointSize} from "./point.js";
+import {abs, atan2, cos, pi, tau, sin, sqrt, max, min} from "./math.js";
+
+export default function(x, y, size) {
+  var defined = constant(true),
+      context = null,
+      ready,
+      ready2,
+      scaleRatio_min = 1,
+      ct1, ct2,
+      px1, py1, pr1,
+      arp = [],
+      p;
+
+  x = typeof x === "function" ? x : (x === undefined) ? pointX : constant(x);
+  y = typeof y === "function" ? y : (y === undefined) ? pointY : constant(y);
+  size = typeof size === "function" ? size : (size === undefined) ? pointSize : constant(size);
+
+  function cross(x1, y1, x2, y2) {
+    return x1 * y2 - x2 * y1;
+  }
+
+  function isIntersect(x1, y1, x2, y2, x3, y3, x4, y4) {
+    // rapid repulsion.
+    if (max(x3, x4) < min(x1, x2) || max(y3, y4) < min(y1, y2) || max(x1, x2) < min(x3, x4) || max(y1, y2) < min(y3, y4)) {
+      return false;
+    }
+
+    // straddle experiment.
+    if (cross(x1 - x4, y1 - y4, x3 - x4, y3 - y4) * cross(x2 - x4, y2 - y4, x3 - x4, y3 - y4) > 0 ||
+       cross(x3 - x2, y3 - y2, x1 - x2, y1 - y2) * cross(x4 - x2, y4 - y2, x1 - x2, y1 - y2) > 0) {
+      return false;
+    }
+    return true;
+  }
+
+  function intersectPoint(x1, y1, x2, y2, x3, y3, x4, y4) {
+    var d1 = abs(cross(x4 - x3, y4 - y3, x1 - x3, y1 - y3)),
+        d2 = abs(cross(x4 - x3, y4 - y3, x2 - x3, y2 - y3)),
+        t;
+
+    if (!(d1 || d2)) {
+      return {
+        x: x2,
+        y: y2
+      };
+    }
+    else {
+      t = d1 / (d1 + d2);
+      return {
+        x: x1 + (x2 - x1) * t,
+        y: y1 + (y2 - y1) * t
+      };
+    }
+  }
+
+  function commonTangent(x1, y1, w1, x2, y2, w2) {
+    var r1 = w1 / 2,
+        r2 = w2 / 2,
+        alpha, // the smallest angle in right-angled trapezoid.
+        alpha_x,
+        alpha_y,
+        beta, // the slope of line segment.
+        cos_x,
+        sin_y;
+
+    alpha_x = abs(r2 - r1);
+    alpha_y = sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2) - alpha_x * alpha_x);
+    alpha = atan2(alpha_y, alpha_x);
+    beta = atan2(y2 - y1, x2 - x1);
+
+    if (r1 < r2) {
+      cos_x = -cos(alpha + beta);
+      sin_y = -sin(alpha + beta);
+
+      return {
+        sg1_x: x1 + r1 * cos_x,
+        sg1_y: y1 + r1 * sin_y,
+        sg2_x: x2 + r2 * cos_x,
+        sg2_y: y2 + r2 * sin_y,
+        angle: alpha + beta - pi
+      };
+    }
+    else {
```

Comment:
No need for this `else`, as the `if` returns. Can remove the brackets and un-indent here.